compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java

```diff
@@ -582,6 +584,8 @@ public void visit(BLangWhile whileNode) {
             this.statementReturnsPanicsOrFails = prevStatementReturnsPanicsOrFails;
             this.continueAsLastStatement = prevContinueAsLastStatement;
             this.breakAsLastStatement = prevBreakAsLastStatement;
+        } else {
+            this.statementReturnsPanicsOrFails = true;
```

Comment:
Not related to this PR, but what do we mean by "Fails" here?

Reply:
when there's a `fail-stmt`